### ISSUE TITLE
Cannot compile GLSL compute shader without HLSL support enabled

### ISSUE THREAD

**Issue by R-Michelsen** [KhronosGroup/glslang#1989]
I've encountered a strange bug (I think) in the library. When I build with HLSL support it works as expected, but if I build without HLSL support I get a compilation error on a particular compute shader. (The shader is written in GLSL, so no HLSL support should be necessary)

I've put the source code used for compilation below, the actual shader source is attached as a file. 

```C
glslang::InitializeProcess();
std::ifstream file("ifft_comp.glsl");
std::string InputGLSL((std::istreambuf_iterator<char>(file)),
			std::istreambuf_iterator<char>());
glslang::TShader Shader(EShLanguage::EShLangCompute);
char const* const str = InputGLSL.c_str();
Shader.setStrings(&str, 1);
auto messages = static_cast<EShMessages>(EShMsgSpvRules | EShMsgVulkanRules);
auto success = Shader.parse(&DDefaultTBuiltInResource, 460, ECoreProfile, false, true, messages);
```
[ifft_comp.glsl.txt](https://github.com/KhronosGroup/glslang/files/3887208/ifft_comp.glsl.txt)

This bug only appears to be present in the master branch as of Nov 25. I cannot reproduce the behaviour on the stable release from Nov 1.

**Comment by johnkslang** [comment]
#2016 reports a similar problem.

I'm hoping it is the same situation where something is broken due to compiling headers with different settings, and everything just needs a recompile with the same settings.

Now, current top of master doesn't include (that I can see) any data structure differences based on ENABLE_HLSL, so that should no longer be the problem.  There are some member function differences.  Would a compiler remove data fields and change offsets within a struct due to member functions being missing?

Either way, this was supposed to be fixed around then by eliminating this source of issue, and I hope it's just that not all code including the headers has been recompiled yet.

**Comment by johnkslang** [comment]
Now I think it is the uninitialized member found by @stuartcarnie in #2020, with fix going in now as #2021.
